frontend/src/components/features/conversation-panel/microagents-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { ChevronDown, ChevronRight } from "lucide-react";
+import { BaseModalTitle } from "#/components/shared/modals/confirmation-modals/base-modal";
+import { ModalBackdrop } from "#/components/shared/modals/modal-backdrop";
+import { ModalBody } from "#/components/shared/modals/modal-body";
+import { I18nKey } from "#/i18n/declaration";
+import { useConversationMicroagents } from "#/hooks/query/use-conversation-microagents";
+
+interface MicroagentsModalProps {
+  onClose: () => void;
+  conversationId: string | undefined;
+}
+
+export function MicroagentsModal({
+  onClose,
+  conversationId,
+}: MicroagentsModalProps) {
+  const { t } = useTranslation();
+  const [expandedAgents, setExpandedAgents] = useState<Record<string, boolean>>(
+    {},
+  );
+
+  const { microagents, isLoading, isError } = useConversationMicroagents({
+    conversationId,
+    enabled: true,
+  });
```

Comment:
```suggestion
  const {
    data: microagents,
    isLoading,
    isError,
  } = useConversationMicroagents({
    conversationId,
    enabled: true,
  });
```

Required change after `frontend/src/hooks/query/use-conversation-microagents.ts` update